src/message.cpp

```diff
@@ -619,12 +616,12 @@ void message::parse_header_line(const string& header_line)
     else if (iequals(header_name, MESSAGE_ID_HEADER))
     {
         auto ids = parse_many_ids(header_value);
-		if (_strict_mode && ids.empty())
-			_message_id = {};
-		else if (!ids.empty())
-			_message_id = ids[0];
-		else
-			throw message_error("Parsing failure of the message ID.");
+        if (!ids.empty())
+        {
+            if (ids.size() > 1)
+                throw message_error("Parsing failure of the message ID.");
```

Comment:
Thanks for your PR. It does look like it solves the issue, but there's one minor concern.
The old behaviour was to allow an email contain multiple IDs in the `Message-ID` header:
https://github.com/karastojko/mailio/blob/f82f7e8e0d9f1c2794a6960e409ddc4899e4881d/src/message.cpp#L614-L618
i.e., no exceptions were thrown in this case, instead, the code just used the first specified ID as the message ID.
The new version throws an exception if `ids`'s size is greater than one. Could you please remove this as this seems to be a breaking change.

Reply:
Done